object-handlers_test.go

```diff
@@ -0,0 +1,358 @@
+		metaData      map[string]string
+	}{
+		// case - 1.
+		{bucketName, objectName, int64(len(bytesData[0].byteData)), bytesData[0].byteData, make(map[string]string)},
+	}
+	// iterate through the above set of inputs and upload the object.
+	for i, input := range putObjectInputs {
+		// uploading the object.
+		_, err = obj.PutObject(input.bucketName, input.objectName, input.contentLength, bytes.NewBuffer(input.textData), input.metaData)
+		// if object upload fails stop the test.
+		if err != nil {
+			t.Fatalf("Put Object case %d:  Error uploading object: <ERROR> %v", i+1, err)
+		}
+	}
+
+	// test cases with inputs and expected result for GetObject.
+	testCases := []struct {
+		bucketName string
+		objectName string
+		byteRange  string // range of bytes to be fetched from GetObject.
+		accessKey  string
+		secretKey  string
+		// expected output.
+		expectedContent    []byte // expected response body.
+		expectedRespStatus int    // expected response status body.
+	}{
+		// Test case - 1.
+		// Fetching the entire object and validating its contents.
+		{
+			bucketName:         bucketName,
+			objectName:         objectName,
+			byteRange:          "",
+			accessKey:          credentials.AccessKeyID,
+			secretKey:          credentials.SecretAccessKey,
+			expectedContent:    bytesData[0].byteData,
+			expectedRespStatus: http.StatusOK,
+		},
+		// Test case - 2.
+		// Case with non-existent object name.
+		{
+			bucketName:         bucketName,
+			objectName:         "abcd",
+			byteRange:          "",
+			accessKey:          credentials.AccessKeyID,
+			secretKey:          credentials.SecretAccessKey,
+			expectedContent:    encodeResponse(getAPIErrorResponse(getAPIError(ErrNoSuchKey), getGetObjectURL("", bucketName, "abcd"))),
+			expectedRespStatus: http.StatusNotFound,
+		},
+		// Test case - 3.
+		// Requesting from range 10-100.
+		{
+			bucketName:         bucketName,
+			objectName:         objectName,
+			byteRange:          "bytes=10-100",
+			accessKey:          credentials.AccessKeyID,
+			secretKey:          credentials.SecretAccessKey,
+			expectedContent:    bytesData[0].byteData[10:101],
+			expectedRespStatus: http.StatusPartialContent,
+		},
+		// Test case - 4.
+		// Test case with invalid range.
+		{
+			bucketName:         bucketName,
+			objectName:         objectName,
+			byteRange:          "bytes=-0",
+			accessKey:          credentials.AccessKeyID,
+			secretKey:          credentials.SecretAccessKey,
+			expectedContent:    encodeResponse(getAPIErrorResponse(getAPIError(ErrInvalidRange), getGetObjectURL("", bucketName, objectName))),
+			expectedRespStatus: http.StatusRequestedRangeNotSatisfiable,
+		},
+		// Test case - 5.
+		// Test case with byte range exceeding the object size.
+		// Expected to read till end of the object.
+		{
+			bucketName:         bucketName,
+			objectName:         objectName,
+			byteRange:          "bytes=10-1000000000000000",
+			accessKey:          credentials.AccessKeyID,
```

Comment:
This is incorrect style, we shouldn't put everything into a single struct like this. Each requests don't have new access keys. We should split the cases with or without signed requests in different tests. 

This way its easy to fix bugs and run only those tests selectively.